src/go/rpk/pkg/cli/cmd/cluster/config/importconfig/import.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package importconfig
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
+	yaml "gopkg.in/yaml.v3"
+)
+
+func ImportConfig(
+	client *admin.AdminAPI,
+	filename string,
+	old_config admin.Config,
+	schema admin.ConfigSchema,
+	all bool,
+) (err error) {
+	readback_bytes, err := ioutil.ReadFile(filename)
+	out.MaybeDie(err, "Error reading file %s: %v", filename, err)
+	var readbackConfig admin.Config
+	err = yaml.Unmarshal(readback_bytes, &readbackConfig)
+	out.MaybeDie(err, "Error parsing edited config: %v", err)
+
+	// Calculate deltas
+	upsert := make(map[string]interface{})
+	remove := make([]string, 0)
+	for k, v := range readbackConfig {
+		oldVal, haveOldVal := old_config[k]
+		if meta, ok := schema[k]; ok {
+			// For numeric types need special handling because
+			// yaml encoding will see '1' as an integer, even
+			// if it is given as the value for a floating point
+			// ('number') config property, and vice versa.
+			if meta.Type == "integer" {
+				switch v_float := v.(type) {
+				case float64:
+					v = int(v_float)
+				}
+
+				if oldVal != nil {
+					oldVal = int(oldVal.(float64))
+				}
+			} else if meta.Type == "number" {
+				switch x := v.(type) {
+				case int:
+					v = float64(x)
+				}
+			} else if meta.Type == "array" && meta.Items.Type == "string" {
+				if v != nil {
+					v = loadStringArray(v.([]interface{}))
+				}
+				if oldVal != nil {
+					oldVal = loadStringArray(oldVal.([]interface{}))
+				}
+			}
+
+			// For types that aren't numberic or array, pass them through as-is
+		}
+
+		if haveOldVal {
+			// If value changed, add it to list of updates
+			// DeepEqual because values can be slices
+			if !reflect.DeepEqual(oldVal, v) {
+				log.Infof("Changed property %s: %v -> %v", k, oldVal, v)
```

Comment:
I think we probably want to build a list of changes / new properties / cleared properties / static properties and print them at the end only if the patch is successful. Potentially:
```
type delta struct {
  Key string
  Prior string
  Current string
}
```
with output
```
KEY    PRIOR    CURRENT
foo    -        3
bar    3        -
baz    5        3
```
(alternatively, empty columns in prior for new, in current for deleted)

Reply:
No objection to switching to a tabular format. I think it's useful to output the table before the API call, so that on an server side validation failure the user can see what their edits were -- then they get a final line that tells them whether the update succeeded.